gax/test/test-application/package.json

```diff
@@ -25,10 +25,11 @@
   },
   "dependencies": {
     "@grpc/grpc-js": "~1.6.0",
-    "google-gax": "file:./google-gax.tgz",
+    "google-gax": "file:google-gax.tgz",
+    "protobufjs": "^7.0.0",
+    "pumpify": "^2.0.1",
```

Comment:
What are your thoughts on using [`Readable.from`](https://nodejs.org/api/stream.html#streamreadablefromiterable-options) rather than `pumpify`?

Example:
```ts
async function* concatStreams(streams: ReadableStream[]) {
  for (const stream of streams) {
    yield* stream;
  }
}

const readable = Readable.from(concatStreams());
```

Reply:
Great question- generally I prefer to use `pipeline` instead of `pumpify`, but I explicitly want pumpify as part of the test application because I know it's used in at least one of the client libraries and was related to the issue that this PR is fixing. I will add comments around that though, and if you want to see a test including Readable.from as an option, I can do that too

Reply:
For now, I have reorganized all of the pumpify and all of the pipeline ones (I know, the organization of test application is a lil chaotic at this point) and I added comments about why they're both there. Lmk if you want me to add Readable.from tests also

Reply:
Ideally, we'd want to recommend whoever's using `pumpify` off of it, as itself is pretty flaky. That would likely be a semver-major issue though.

Reply:
@danielbankhead let me know what the preferred action within gax is - I can open an issue in the affected client library to stop using pumpify

Reply:
I created an issue to migrate the stakeholder off of pumpify, and I'm going to merge this for now. If you want to create an issue to have it removed later please feel free to!